apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -24,72 +42,82 @@ const maybeAlert = (alerts: Alert[]): ReactElement<HTMLElement> | null => {
+}: Props): ReactElement<HTMLElement> | null => {
+  const routeType = lineDiagram[0].route_stop.route!.type;
+
+  return (
+    <>
+      <h3>
+        {routeType === 0 || routeType === 1 || routeType === 2
+          ? "Stations"
+          : "Stops"}
+      </h3>
+      {lineDiagram.map(
+        ({ route_stop: routeStop, alerts: stopAlerts }: LineDiagramStop) => (
+          <div key={routeStop.id} className="m-schedule-line-diagram__stop">
+            <div className="m-schedule-line-diagram__card-left">
+              <div className="m-schedule-line-diagram__stop-name">
+                {maybeAlert(stopAlerts)}
+                <a href={`/stops/${routeStop.id}`}>
+                  <h4>{routeStop.name}</h4>
+                </a>
+              </div>
+              <div className="m-schedule-line-diagram__connections">
+                {filteredConnections(routeStop.connections).map(
+                  (route: Route) =>
+                    route.type === 3 ? (
+                      <a href={`/schedules/${route.id}/line`}>
+                        <span
+                          key={route.id}
+                          className={`c-icon__bus-pill--small m-schedule-line-diagram__connection ${busBackgroundClass(
+                            route
+                          )}`}
+                        >
+                          {route.name}
+                        </span>
+                      </a>
+                    ) : (
+                      <a href={`/schedules/${route.id}/line`}>
+                        <span
+                          key={route.id}
+                          className="m-schedule-line-diagram__connection"
+                        >
+                          {modeIcon(route.id)}
+                        </span>
+                      </a>
+                    )
+                )}
+              </div>
             </div>
             <div>
-              {/* eslint-disable react/no-array-index-key */
-              stopData.map((stop: StopData, idx: number) => (
-                <div key={idx}>
-                  {stop.branch ? `${stop.branch}, ` : ""}
-                  {stop.type}
-                </div>
-              ))}
-              {/* eslint-enable react/no-array-index-key */}
+              <div className="m-schedule-line-diagram__features">
+                {routeStop.stop_features.includes("parking_lot")
+                  ? parkingIcon(
+                      "c-svg__icon-parking-default m-schedule-line-diagram__feature-icon"
+                    )
+                  : null}
+                {routeStop.stop_features.includes("access")
+                  ? accessibleIcon(
+                      "c-svg__icon-acessible-default m-schedule-line-diagram__feature-icon"
+                    )
+                  : null}
+                {routeStop.route!.type === 2 && routeStop.zone && (
+                  <span className="c-icon__cr-zone m-schedule-line-diagram__feature-icon">{`Zone ${
```

Comment:
There seems to be a greedy CSS issue happening here which is crushing Zone pills into squares. See image below:

![image](https://user-images.githubusercontent.com/688435/68332955-a5d49e00-00a5-11ea-9344-6943380c19e9.png)

Reply:
I think the font size for the zone needs to come down, too.